File: src/main/scala/classtest/SuperTest.scala
package classtest.supertest

object SuperTest extends App {
  private val son = new Son(10)
  println(son.getFatherAttr)
  println(son.getSonAttr)
//  println(son.attr)
}
class Father(val attr:Int = 0){
  println(s"father $attr")
  def getFatherAttr = attr*2
}
//**注意不能通过super关键字获取父类字段编译会报错
//可以同时存在
class Son(attr:Int) extends Father(30) {
  println(s"son $attr")

  def getSonAttr = super.getFatherAttr * 3

}

File: src/main/java/supertest/SuperTest.java
package supertest;

public class SuperTest {
    public static void main(String[] args) {
        Son son1 = new Son("hello");
        Father son2 = new Son("hello");
        System.out.println(son1.getName());
        System.out.println(son2.getName());
    }

}

class Father {
    private String name;

    public Father(){}
    public Father(String name) {
        this.name = name;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }
}

class Son extends Father{
    private String name;
    public Son(String name){
        super(name);
        this.name=super.getName()+name;
    }

    @Override
    public String getName() {
        return name;
    }

    @Override
    public void setName(String name) {
        this.name = name;
    }
}
